### DISCUSSION TITLE
Bot heath check periodically

### DISCUSSION THREAD

**Discussion by juanblasmdq** [telegraf/telegraf#1761] [Q&A] [answered]
Hi everyone!

I would like to know if anyone has ever implemented something similar to what I describe below and can give me a hint on how to proceed. I would like to use Github actions or other service to periodically check the status of my bot and, if status is Not ok, send me an email or mobile alert

I imagine something like the following:
- Generate an ip/http public address and expose the bot to that address, so I can make a GET request to check bot's status (if ok reply code 200)
- Run a cron job to make a GET request every 30min
- Cron job sends me an email if status is not ok

Any advice / example will be appreciated since I'm pretty new to programming (just learning as a hobby)

Thanks and regards,
JBC

**Comment by MKRhere** [comment]
This looks fine already. Running the cronjob may be the only issue, since now you need two services alive (the bot itself and the cronjob healthcheck service). You may want to use any thirdparty http monitoring system (uptimerobot, etc.) instead.

**Comment by juanblasmdq** [comment]
Hi Muthu, thanks for the reply! Do you have any example on how to do what I describe in the first bullet?

Ps: I was ban from the telegram chat by telegram spam bot when trying to post a nodejs error log, lol

**Reply by MKRhere** [reply]
Just have any route that returns 200. If you use Express:

```TS
app.get("/health", (req, res) => res.send({ status: "ok" }}));
```

You can do this with your framework of choice.

> ban from the telegram chat by telegram spam bot

I noticed today that it has been flagging many false positives. I have turned off the new aggressive spam filter. Were you banned or just had your messages deleted? If your messages were being deleted before, it should be fixed now.

**Comment by juanblasmdq** [comment, marked as answer]
Oh, I thought it was way more difficult to integrate telegraf with express. I'll try the way you suggest

Ps: seems that the ban was lifted, thanks

**Reply by MKRhere** [reply]
Example of express integration:
https://github.com/feathers-studio/telegraf-docs/blob/4feffcd893a86ed1fcc761279d17328beb4a2f21/examples/webhook/express.ts

If you don't use ESM, wrap the code in an `async function main () {}` to use await inside.

**Comment by juanblasmdq** [comment]
Many many thanks @MKRhere !
